### PR TITLE
Move user_created_data? from application_controller.rb into comments_helper.rb

### PR DESCRIPTION
**Pull request by asarbu**
# Description 

Move user_created_data? from application_controller.rb into comments_helper.rb

## More Details

Move user_created_data? from application_controller.rb into comments_helper.rb because the action is used only in the helper class.

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1015

# Test Coverage

✅ 
🚫

### CODE REVIEW

File: app/helpers/comments_helper.rb [hunk 1]
```diff
@@ -84,4 +84,20 @@ def comment_for(model_name)
   rescue ActiveRecord::RecordInvalid
     respond_not_saved
   end
+
+  # rubocop:disable MethodLength
+  def user_created_data?(id, data_type)
```
Comment (julianguyen) [inline app/helpers/comments_helper.rb]: So it looks like `user_created_data?` is only used in `comment_deletable?`. I would put a private above this method.

Comment (asarbu) [reply app/helpers/comments_helper.rb]: sure 👍 . Do you think there might be other methods inside this helper that need to be private ?

Comment (julianguyen) [reply app/helpers/comments_helper.rb]: I would double check to make sure, I don't know off the top of my head!

Comment (asarbu) [reply app/helpers/comments_helper.rb]: I looked into the file and moved the methods that were privat under that scope. Let me know what you think

**Pull request merged**